### PR TITLE
Worker conditions

### PR DESCRIPTION
**Pull request by simonb83**
Change the way that workers choose which analyses to process to use a filter_function rather than single status.
This allows for more complex conditions.

### CODE REVIEW

File: source/python/run_scraper.py [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
+
+# Filter function for identifying analyses to scrape
+def scraping_filter(query):
+    # Choose either New analyses OR
+    # Analyses where Scraping Failed &
+    # less than 3 scraping attempts &
+    # last scraping attempt greater than 12 hours ago
+    return query.filter((Analysis.status == Status.NEW) |
+                        ((Analysis.status == Status.SCRAPING_FAILED) &
+                         (Analysis.retrieval_attempts < 3) &
```
Comment (WanderingStar) [inline source/python/run_scraper.py]: `3` and `timedelta(hours=12)` seem like they should be in a config somewhere.

**Review by WanderingStar — APPROVED** [review]

File: source/python/run_scraper.py (hunk 1 above)
Comment (simonb83) [reply source/python/run_scraper.py]: Thanks!
Do you think having a config.py file somewhere and importing that is the right way to go, or would you recommend a different approach?

Comment (WanderingStar) [reply source/python/run_scraper.py]: I'd probably start by making these file-local constants, but yeah, eventually a general config.py or config.json makes sense to me. There's no _pressing_ need to split this out, it's just the kind of thing that seems like someone will want to tune at some point.

File: source/python/run_scraper.py [hunk 2]
```diff
@@ -9,6 +9,10 @@
 from idetect.worker import Worker
 
 
+MAX_RETRIEVAL_ATTEMPTS = 3
+TIME_BETWEEN_ATTEMPTS = 12
```
Comment (WanderingStar) [inline source/python/run_scraper.py]: HOURS_BETWEEN_ATTEMPTS? Or a comment mentioning that 12 is in hours?

**Pull request merged**